server/src/dependency/heed.rs

```diff
@@ -52,14 +49,13 @@ pub fn state_query(genesis_config: &GenesisConfig) -> StateQueries {
 
 pub fn on_tx_batch<
     A: State,
-    B: NewPayloadId,
     C: BlockHash,
     D: BlockRepository<Storage = SharedStorage>,
     E: BaseGasFee,
     F: CreateL1GasFee,
     G: CreateL2GasFee,
     H: BaseTokenAccounts,
->() -> OnTxBatch<A, B, C, D, E, F, G, H> {
+>() -> OnTxBatch<A, C, D, E, F, G, H> {
```

Comment:
Instead of removing the `B`, we can shift them and go from `A .. G`. Same applies to the ones below.